incubator/orm/indexer.go

```diff
@@ -148,3 +148,19 @@ func makeIndexPrefixScanKey(indexKey []byte, rowID uint64) []byte {
 	binary.BigEndian.PutUint64(res[n:], rowID)
 	return res
 }
+
+// pruneEmptyKeys drops any empty key from IndexerFunc f returned
+func pruneEmptyKeys(f IndexerFunc) IndexerFunc {
+	return func(v interface{}) ([][]byte, error) {
+		keys, err := f(v)
+		if err != nil {
+			return nil, err
+		}
+		for i := 0; i < len(keys); i++ {
+			if len(keys[i]) == 0 {
+				keys = append(keys[0:i], keys[i+1:]...)
```

Comment:
I'm a bit wary of these types of operations in general. I'm assuming `append` doesn't allocate here because the slice is getting smaller if anything. The thing that seems potentially problematic is that it seems like you're mutating a slice while iterating over it. So whenever you hit an empty key, the `i+1`th element now becomes the `i`th element, but since you've already checked the `i`th element, I think the `i+1`th element wouldn't actually get checked. Am I right about that or am I missing something?

Also, I don't generally like to call `len` in the test condition of a for loop as it will get evaluated everything, here likely zero cost but I'm remembering `strlen` from C.

Reply:
np, I have redesigned the func.